ZipBuilder/Sources/ZipBuilder/LaunchArgs.swift

```diff
@@ -0,0 +1,240 @@
+        "containing all required files to build frameworks.")
+    }
+
+    templateDir = URL(fileURLWithPath: templatePath)
+
+    // Parse the path to CoreDiagnostics.framework.
+    guard let diagnosticsPath = defaults.string(forKey: Key.coreDiagnosticsDir.rawValue) else {
+      LaunchArgs.exitWithUsageAndLog("Missing required key: `\(Key.coreDiagnosticsDir)` for the " +
+        "path to the CoreDiagnostics framework.")
+    }
+
+    coreDiagnosticsDir = URL(fileURLWithPath: diagnosticsPath)
+
+    // Parse the existing versions key.
+    if let existingVersions = defaults.string(forKey: Key.existingVersions.rawValue) {
+      let url = URL(fileURLWithPath: existingVersions)
+      guard fileChecker.fileExists(atPath: url.path) else {
+        LaunchArgs.exitWithUsageAndLog("Could not parse \(Key.existingVersions) key: value " +
+          "passed in is not a file URL or the file does not exist. Value: \(existingVersions)")
+      }
+
+      allSDKsPath = url.standardizedFileURL
+    } else {
+      // No argument was passed in.
+      allSDKsPath = nil
+    }
+
+    // Parse the current releases key.
+    if let currentRelease = defaults.string(forKey: Key.releasingSDKs.rawValue) {
+      let url = URL(fileURLWithPath: currentRelease)
+      guard fileChecker.fileExists(atPath: url.path) else {
+        LaunchArgs.exitWithUsageAndLog("Could not parse \(Key.releasingSDKs) key: value passed " +
+          "in is not a file URL or the file does not exist. Value: \(currentRelease)")
+      }
+
+      currentReleasePath = url.standardizedFileURL
+    } else {
+      // No argument was passed in.
+      currentReleasePath = nil
+    }
+
+    // Parse the output directory key.
+    if let outputPath = defaults.string(forKey: Key.outputDir.rawValue) {
+      let url = URL(fileURLWithPath: outputPath)
+      guard fileChecker.directoryExists(at: url) else {
+        LaunchArgs.exitWithUsageAndLog("Could not parse \(Key.outputDir) key: value " +
+          "passed in is not a file URL or the directory does not exist. Value: \(outputPath)")
+      }
+
+      outputDir = url.standardizedFileURL
+    } else {
+      // No argument was passed in.
+      outputDir = nil
+    }
+
+    // Parse the custom specs key.
+    if let customSpecs = defaults.string(forKey: Key.customSpecRepos.rawValue) {
+      // Custom specs are passed in as a comma separated list of URLs. Split the String by each
+      // comma and map it to URLs. If any URL is invalid, fail immediately.
+      let specs = customSpecs.split(separator: ",").map { (specStr: Substring) -> URL in
+        guard let spec = URL(string: String(specStr)) else {
+          LaunchArgs.exitWithUsageAndLog("Error parsing specs: \(specStr) is not a valid URL.")
+        }
+
+        return spec
+      }
+
+      customSpecRepos = specs
+    } else {
+      // No argument was passed in.
+      customSpecRepos = nil
+    }
+
+    updatePodRepo = defaults.bool(forKey: Key.updatePodRepo.rawValue)
```

Comment:
Where is the default set?  It should be YES.

Reply:
Will add it at the top of this class by registering with User Defaults.